android/java/org/chromium/chrome/browser/billing/InAppPurchaseWrapper.java

```diff
@@ -321,24 +326,44 @@ private void acknowledgePurchase(Context context, Purchase purchase) {
                         }
                         if (billingResult.getResponseCode()
                                 == BillingClient.BillingResponseCode.OK) {
-                            BraveVpnPrefUtils.setSubscriptionPurchase(true);
-                            if (activity != null) {
-                                BraveVpnUtils.openBraveVpnProfileActivity(activity);
-                                BraveVpnUtils.showToast(activity.getResources().getString(
-                                        R.string.subscription_consumed));
+                            if (isVPNProduct) {
+                                BraveVpnPrefUtils.setSubscriptionPurchase(true);
+                                if (activity != null) {
+                                    BraveVpnUtils.openBraveVpnProfileActivity(activity);
+                                }
+                            } else if (isLeoProduct) {
+                                BraveLeoPrefUtils.setIsSubscriptionActive(true);
                             }
+                            showToast(context.getResources().getString(
+                                    R.string.subscription_consumed));
                         } else {
-                            BraveVpnUtils.showToast(
+                            showToast(
                                     context.getResources().getString(R.string.fail_to_aknowledge));
                         }
                     });
                 }
             });
         } else {
-            BraveVpnPrefUtils.setSubscriptionPurchase(true);
+            if (isVPNProduct) {
+                BraveVpnPrefUtils.setSubscriptionPurchase(true);
+            } else if (isLeoProduct) {
```

Comment:
is there any purpose for this empty block?

Reply:
not really, I need to set that purchase is active in that case, corrected.